packages/datadog-instrumentations/src/url.js

```diff
@@ -0,0 +1,85 @@
+'use strict'
+
+const { addHook, channel } = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+const names = ['url', 'node:url']
+
+const parseFinishedChannel = channel('datadog:url:parse:finish')
+const urlGetterChannel = channel('datadog:url:getter:finish')
+const instrumentedGetters = ['host', 'origin', 'hostname']
+
+addHook({ name: names }, function (url) {
+  shimmer.wrap(url, 'parse', (parse) => {
+    return function wrappedParse () {
+      const parsedValue = parse.apply(this, arguments)
+      if (!parseFinishedChannel.hasSubscribers) return parsedValue
+
+      parseFinishedChannel.publish({
+        input: arguments[0],
+        parsed: parsedValue,
+        isURL: false
+      })
+
+      return parsedValue
+    }
+  })
+
+  const URLPrototype = url.URL.prototype.constructor.prototype
+  instrumentedGetters.forEach(property => {
+    const originalDescriptor = Object.getOwnPropertyDescriptor(URLPrototype, property)
+
+    if (originalDescriptor?.get) {
+      const newDescriptor = {
+        ...originalDescriptor,
+        get: function () {
+          const result = originalDescriptor.get.apply(this, arguments)
+          if (!urlGetterChannel.hasSubscribers) return result
+
+          const context = { urlObject: this, result, property }
+          urlGetterChannel.publish(context)
+
+          return context.result
+        }
+      }
+
+      Object.defineProperty(URLPrototype, property, newDescriptor)
+    }
+  })
+
+  shimmer.wrap(url, 'URL', (URL) => {
+    return class extends URL {
+      constructor () {
+        super(...arguments)
+
+        if (!parseFinishedChannel.hasSubscribers) return
+
+        parseFinishedChannel.publish({
+          input: arguments[0],
+          base: arguments[1],
```

Comment:
I put the same comment in Ilyas' PR but why using `arguments[x]` instead of just named arguments ?

Reply:
I think that I am using it because I am already using `arguments` to call to `apply(this, arguments)` or `super(...arguments)`. But yes, it is cleaner if I use `input` and `base` variables.